### PR TITLE
Search bar is now connected to BottomMenu and MoreDetails

### PR DESCRIPTION
**Pull request by AlainJobs**
## Please Test
Because I made some refactoring in important components (MapData.js), please make sure to test the app and try to break the Search and/or MoreDetails components, I believe in you all! 
## Bug fixed
- Issue #210 as well as #208 
## Implementation
Implemented what was asked in issue #161 
### More
- You can search for a building name, full name, address, services, and departments. When selecting any of them, the MoreDetails will display the building information. 
- Adjusted arrowBackButton in DoubleSearch to stay consistent with PreviewDirections.js and Directions.js.
- Added fullName field in MoreDetails.js
- Added 2 FIXME comments regarding some components that trigger bad warnings. 
## Even More
- I upgraded React Native to the most recent version which is [37.0.0](https://blog.expo.io/expo-sdk-37-is-now-available-dd5770f066a6). Expo is now smoother and faster with a slightly new interface. I have some tricks to test your app faster and with fewer tunnel bugs now (pm me if needed)
- I upgraded Expo CLI to the most recent version.
- The two upgrades were successful in Travis and don't forget to do `npm install`.
- I added devtools dependencies that communicate with the Debugger and set up React Native Debugger. So if someone wants to use the debugger, let me know I could help you since it's local (I can't push a debugger remotely).
## Refactoring
- I refactored MapData.js almost completely.
- I refactored MoreDetails a bit.
- I updated Search and DoubleSearch.
## Test
1. Check out to `feature/#166`
2. Make sure to run `npm install` before running the app with _expo start_
3. Search for anything in the Search Bar.
4. Click on the arrow up from bottomMenu.
## We move 🤝

### CODE REVIEW

**Comment by Firassawan** [comment]
Will review first thing in the morning! Thanks Alain!

File: __tests__/unit/Search.test.js [hunk 1]
```diff
@@ -1,10 +1,26 @@
 import React from "react";
 import { Search } from "../../components/Search";
 import renderer from "react-test-renderer";
+import { MapData, RetrieveSearchItems } from "../../components/MapData";
```
Comment (Firassawan) [inline __tests__/unit/Search.test.js:4]: Screen Shot 2020-04-07 at 4 25 55 AM

**Comment by mohanadarafe** [comment]
After running `npm i`, I can't run the app because of this error:

_Unable to resolve "./ScreenOrientation/ScreenOrientation" from "node_modules/expo/build/ExpoLazy.js"
Failed building JavaScript bundle._

**Comment by mohanadarafe** [comment]
The MapData.js refactor is really good. Great documentation & makes the code look cleaner. I ran our Debt tracker on your file & it's all green ✅ 

I tried to break the MoreDetails/Search, I noticed this:
IMG_8046

The addresses should not be in the search bar.

**Comment by AlainJobs** [comment]
Thanks @mohanad, I'll update the PR instructions and I will remove the addresses, it doesn't make sense to have them.

**Comment by csbduzi** [comment]
Amazing Job on the refactoring, a lot code was reduced in MapData and MoreDetails! Very clean! Concise documentation! ✅ 

I tested on my end by selecting items from the search bar and then pressing on the arrow up to go to MoreDetails and it works well. And its still smooth too. 

Also I noticed on that certain MoreDetails screens that the number when left _empty" is not displaying `N/A` anymore.

Image from iOS (3)

**Comment by csbduzi** [comment]
> The MapData.js refactor is really good. Great documentation & makes the code look cleaner. I ran our Debt tracker on your file & it's all green ✅
> 
> I tried to break the MoreDetails/Search, I noticed this:
> IMG_8046
> 
> The addresses should not be in the search bar. Also, the code coverage sky rocketed down from 77% to 62%. I think it's due to route stack. @bkenza can help you get the coverage back up 📈

image

He's right the code coverage has reduced significantly to 62%, however I don't think its because of the route stack since its doesn't have any methods. 

On my end, not sure if it's locally, but the some tests are failing and maybe not covering the code properly

**Comment by bkenza** [comment]
> > The MapData.js refactor is really good. Great documentation & makes the code look cleaner. I ran our Debt tracker on your file & it's all green ✅
> > I tried to break the MoreDetails/Search, I noticed this:
> > IMG_8046
> > The addresses should not be in the search bar. Also, the code coverage sky rocketed down from 77% to 62%. I think it's due to route stack. @bkenza can help you get the coverage back up 📈
> 
> image
> 
> He's right the code coverage has reduced significantly to 62%, however I don't think its because of the route stack since its doesn't have any methods.
> 
> On my end, not sure if it's locally, but the some tests are failing and maybe not covering the code properly

I will take care of the code coverage!

**Comment by badred123** [comment]
I get this error after running `npm start`. I did npm install and it still gives me this error...
92419613_2978261495564464_2199397183994724352_n

**Comment by AlainJobs** [comment]
@csbduzi For the N/A, it should work now.

**Comment by badred123** [comment]
> I get this error after running `npm start`. I did npm install and it still gives me this error...
> 92419613_2978261495564464_2199397183994724352_n

I got it to work !

**Comment by badred123** [comment]
I get this error when one of from/To is "Grey nuns Building" and i click on view routes.
92471387_291552458500978_4412167010658549760_n

**Comment by AlainJobs** [comment]
> I get this error when one of from/To is "Grey nuns Building" and i click on view routes.
> 92471387_291552458500978_4412167010658549760_n

Thank you, I know why!

**Comment by AlainJobs** [comment]
Please guys, for any error, post those in the review section of Github in the appropraite file that broke, its easier to resolve and make the PR less hard to read.

**Pull request merged**